axelar-chains-config/src/utils/verifyContract.js

```diff
@@ -1,5 +1,139 @@
 const { execSync } = require('child_process');
 const { writeFileSync } = require('fs');
+const fs = require('fs');
+const path = require('path');
+const axios = require('axios');
+
+async function getSmallestFile(directoryPath) {
+    const fileNames = await fs.promises.readdir(directoryPath);
+    let smallestFile = null;
+    let smallestSize = Number.MAX_SAFE_INTEGER;
+
+    for (const fileName of fileNames) {
+        const filePath = path.join(directoryPath, fileName);
+        const stats = await fs.promises.stat(filePath);
+
+        if (stats.size < smallestSize) {
+            smallestFile = filePath;
+            smallestSize = stats.size;
+        }
+    }
+
+    return smallestFile;
+}
+
+function getContractFileName(config, address) {
+    for (const [key, currentValue] of Object.entries(config)) {
+        if (currentValue === address) {
+            return key.charAt(0).toUpperCase() + key.slice(1);
+        }
+    }
+
+    return null;
+}
+
+async function findFilePath(startPath, targetFileName) {
+    if (!fs.existsSync(startPath)) {
+        throw new Error(`Start path does not exist: ${startPath}`);
+    }
+
+    const directoriesToSearch = [startPath];
+
+    while (directoriesToSearch.length > 0) {
+        const currentDirectory = directoriesToSearch.pop();
+        const filesAndDirectories = fs.readdirSync(currentDirectory, { withFileTypes: true });
+
+        for (const fileOrDirectory of filesAndDirectories) {
+            const fullPath = path.join(currentDirectory, fileOrDirectory.name);
+
+            if (fileOrDirectory.isDirectory()) {
+                directoriesToSearch.push(fullPath);
+            } else if (fileOrDirectory.isFile() && fileOrDirectory.name === targetFileName) {
```

Comment:
Optional: if we convert `fileOrDirectory.name` and `targetFileName` to lowercase and compare them we should be able to simplify `getContractFileName` by removing slicing and first letter conversion to uppercase, which should increase code readability.